### ISSUE TITLE
Add support for modifying client settings on-the-fly

### ISSUE THREAD

**Issue by ClobberXD** [minetest/minetest#7760]
##### Issue type

 - Feature request

##### Summary

Most other games, be they simple indie titles or the latest AAA titles, allow for changing settings in-game. It is quite boring and unnecessary to have to leave the server, just to adjust a couple of settings. This is especially pain-staking if leaving the server has significant negative effects. e.g. In CTF, a player's inventory is cleared on rejoin, and if one wants to adjust a setting, they'd have to wait till the match is over, which could occasionally take more than 2 hours.

It'd be very useful to be able to access the settings dialog from the pause menu itself, and once the user is done, they should be able to press an "Apply" button to apply all the modified settings on-the-go. Note that the suggested implementation is just an example, and needn't be the most suitable one for MT.

**Comment by SmallJoker** [comment]
Duplicate of #6722, #6922, #3165 and probably more I couldn't find in this single search query.

**Comment by ClobberXD** [comment]
Those issues are related, but aren't the same as this. The first two ask for a more detailed pause menu, and the third asks _only_ for FOV modification on-the-fly. This feature request is about
- The whole "Settings" menu (that's currently accessible from the main menu) to be accessible from the pause menu.
- Some way to apply the modified settings on-the-fly

**Comment by SmallJoker** [comment]
The entire settings menu (assuming you mean the Advanced Settings dialogue) would not be needed for ingame on-the-fly configuration. Only `Graphics`, `Sound` and `Client` would be suited for this task because even in singleplayer the server would not reload the configuration file unless you tell it to do so.
For speed reasons (yes, settings are slow) it's also a good choice to leave some settings cached and not allow changing them directly - reloading each second might be okay.

Taking the advanced settings menu into the game would either require to
1) permanently allow CSMs (we have a restriction for that)
2) or create a new Lua instance which is limited to this feature

Out of the left over configuration possibilities, [this comment](https://github.com/minetest/minetest/issues/6722#issuecomment-348737173) covers pretty much of it. This issue is just another suggestion on how to implement #6722 properly. Please add your suggestion there.

**Comment by ClobberXD** [comment]
> Only Graphics, Sound and Client would be suited for this task because even in singleplayer the server would not reload the configuration file unless you tell it to do so.

Ah ok, thanks for clarifying. I did mean only those. Sorry for yet another dupe :)
